core/src/services/gcs/writer.rs

```diff
@@ -74,8 +69,11 @@ impl GcsWriter {
             _ => Err(parse_error(resp).await?),
         }
     }
+}
 
-    async fn initiate_upload(&self) -> Result<String> {
+#[async_trait]
+impl oio::RangeWrite for GcsWriter {
```

Comment:
We need a test case for it.

Reply:
Already covered by gcs's behavior test. And we can add unit for `RangeWrite` later.